### ISSUE TITLE
suggestion: simplify internal module specifiers

### ISSUE THREAD

**Issue by iuioiua** [denoland/deno#24294]
The internal import map for this codebase is defined in [`tools/core_import_map.json`](https://github.com/denoland/deno/blob/main/tools/core_import_map.json). Currently, each import corresponds to an individual script. This means a new entry is needed for every new script, and when one entry is missing, VSCode IntelliSense doesn't work for symbols from that script.

I suggest replacing these single-script imports with directory imports and changing the prefix used in internal module specifiers. This change would automatically cover almost all internal imports with near-zero maintenance. @bartlomieju told me this would require considerable changes to this repo and [`deno_core`](https://github.com/denoland/deno_core) code and, therefore, likely best done after Deno 2.

Please excuse me if I've overlooked some obvious drawbacks to this approach. I have little context on why internal module specifiers were done this way in the first place.

### Currently

https://github.com/denoland/deno/blob/88e3f465d3a95db277ac7e685a0ed239958f8e6e/tools/core_import_map.json#L55-L60

https://github.com/denoland/deno/blob/88e3f465d3a95db277ac7e685a0ed239958f8e6e/ext/node/polyfills/_brotli.js#L29-L32

### Suggestion

```jsonc
// tools/core_import_map.json
{
  "imports": {
    "ext/": "../ext/",
    "ext/node/": "../ext/node/polyfills",
    "ext/runtime/": "../runtime/js/",
    "node:assert": "../ext/node/polyfills/assert.ts",
    "node:assert/strict": "../ext/node/polyfills/assert/strict.ts"
    // ...
  }
}
```

```js
// ext/node/polyfills/_brotli.js
// ...
import { zlib as constants } from "ext/node/internal_binding/constants.ts";
import { TextEncoder } from "ext/web/08_text_encoding.js";
import { Transform } from "node:stream";
import { Buffer } from "node:buffer";
// ...
```

**Comment by lucacasonato** [comment]
I don't think this would require deno_core changes if the only thing you want to change is the import map.

**Comment by iuioiua** [comment]
We'd have to change the import map as well as module specifiers in TS/JS files.

**Comment by lucacasonato** [comment]
Oh I see - we could just change the specifiers to `ext:/...` - deno_core could likely support that easially

**Comment by iuioiua** [comment]
I tried that, and IntelliSense didn't work (see [here](https://github.com/denoland/deno/compare/main...iuioiua:deno:simplify-internal-imports)). Perhaps I'm missing something obvious, but I can't find documentation on directory-based imports with prefixes like `ext:`.

**Comment by bartlomieju** [comment]
Not gonna do that after all. Too many changes for not much gain.
